dashboard/lib/queries/child_account.rb

```diff
@@ -20,4 +20,23 @@ def self.expired_accounts(scope: User.all, expiration_date: 7.days.ago)
         expiration_date
       )
   end
+
+  # Retrieve the most recent ParentalPermissionRequest for a user
+  def self.latest_permission_request(user)
+    ParentalPermissionRequest.where(user: user).order(updated_at: :desc).limit(1).first
+  end
+
+  # Find the US state that the student most likely resides in, based on the state
+  # associated with their most recent teacher's school.
+  def self.teacher_us_state(user)
+    return nil unless user.student?
+
+    latest_student_section = user.sections_as_student.order(created_at: :desc).first
+    return nil unless latest_student_section
+
+    latest_teacher_school_info = UserSchoolInfo.where(user_id: latest_student_section.user_id).order(start_date: :desc).first
+    return nil unless latest_teacher_school_info
+
+    latest_teacher_school_info.school_info&.state
```

Comment:
Should we also check the country in the `school_info`? I'm worried there might be a different country which has `CO` has the state abbreviation.

Reply:
I just added this check but then realized it isn't necessary. The `school_infos` model already has validation which doesn't allow `state` to be populated unless the `country` is the US. The `state` field here is really intended as a US state only, not a locality in a non-US country.

Reply:
I can see that not being true in the future, if you'd like to be pedantic.